packages/monaco/src/browser/monaco-text-model-service.ts

```diff
@@ -62,6 +63,34 @@ export class MonacoTextModelService implements monaco.editor.ITextModelService {
     @inject(ILogger)
     protected readonly logger: ILogger;
 
+    @inject(ApplicationServer)
+    protected readonly applicationServer!: ApplicationServer;
+
+    @postConstruct()
+    public init(): void {
+        let isWindowsBackend = false;
+
+        this.applicationServer.getBackendOS().then(os => {
+            isWindowsBackend = os === OS.Type.Windows;
```

Comment:
Is `false` a good default or is it better to use the same mechanism you did with the deferred value on the last iteration of your changes? This time the deferred would stay protected and only used within this class, and it would make sure that we use the right value for the given backend?

Reply:
@federicobozzini I just pushed what I suggested to your branch 👍

Reply:
@marechal-p  Sorry for the late reply. I did some investigation about adding a deferred barrier to the object initialization, but what I've noticed is that with that a long `configure` step might create some issues with the Theia plugins initialization. It's something that I was still investigating so it might be unrelated to this change.

Reply:
I'll wait for your feedback then, thank you for the info!